lib/active_record_view_model/update_data.rb

```diff
@@ -203,11 +247,23 @@ def parse(hash_data, valid_reference_keys)
             end
 
             if association_data.collection?
-              unless value.is_a?(Array)
-                raise ViewModel::DeserializationError.new("Could not parse non-array collection association")
-              end
 
-              associations[name] = value.map { |child_hash| parse_association.(child_hash) }
+              associations[name] =
+                case
+                when value.is_a?(Array)
+                  children = value.map { |child_hash| parse_association.(child_hash) }
+                  [{ :op => :replace, :values => children }]
```

Comment:
replace is necessarily incompatible with any other updates: is it fair to consider it of the same essential type and have it in an array?